site/content/integrate/slash-commands/custom/_index.md

```diff
@@ -0,0 +1,137 @@
+
+2. Select **Add Slash Command**; the **Add** dialog will appear. Use the following guidelines to configure the slash command:
+
+    - Set the **Title** and **Description** for the command.
+    - Set the **Command Trigger Word**. The trigger word must be unique and cannot begin with a slash or contain any spaces. It also cannot be one of the [built-in commands]({{< ref "/integrate/slash-commands/built-in" >}}).
+    - Set the **Request URL** and **Request Method**. The request URL is the endpoint that Mattermost hits to reach your application, and the request method is either `POST` or `GET` and specifies the type of request sent to the request URL.
+    - (_Optional_) Set the **Response Username** and **Response Icon** the command will post messages as in Mattermost. If not set, the command will use your username and profile picture.
+    - (_Optional_) Select the **Autocomplete** option to include the slash command in the command autocomplete list, displayed when typing ``/`` in an empty input box. Use it to make your command easier to discover by your teammates. You can also provide a hint listing the arguments of your command and a short description displayed in the autocomplete list.
+
+      {{<note "Note:">}}
+   [Enable integrations to override usernames](https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-usernames) must be set to `true` in `config.json` to override usernames. Enable them from **System Console > Integrations > Integration Management**, or ask your System Admin to do so. If not enabled, the username is set to `webhook`.
+
+   Similarly, [Enable integrations to override profile picture icons](https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-profile-picture-icons) must be set to `true` in `config.json` to override profile picture icons. Enable them from **System Console > Integrations > Integration Management**, or ask your System Admin to do so. If not enabled, the icon of the creator of the webhook URL is used to post messages.
+      {{</note>}}
+
+3. Select **Save**. On the next page, copy the **Token** value. This will be used in a later step.
+
+   ![image](slash_commands_token.png)
+
+4. Next, write your external application. Include a function which receives HTTP `POST` or HTTP `GET` requests from Mattermost. The request will look something like this:
+
+    ```http request
+    POST /slash-command HTTP/1.1
+    Host: example.com
+    Accept: application/json
+    Accept-Encoding: gzip
+    Authorization: Token qzgakf1nx3yt9dr4n8585ihbxy
+    Content-Length: 313
+    Content-Type: application/x-www-form-urlencoded
+    User-Agent: Mattermost-Bot/1.1
+    
+    channel_id=jux16pkewjrkfj3ehep1psxyxc&
+    channel_mentions=["saepe-5", "aut-8"]&
+    channel_mentions_ids=["r3j6sby343fpfdxcbwqg95rfsa", "ehjj46yk7ifptr5bpfb966s6mc"]&
+    channel_name=town-square&
+    command=%2Ftest&
```

Comment:
Wouldn't it make more sense to show us what the `/weather toronto week` request would look like under the hood?

Reply:
Great idea!